atlp-rwanda/e-comm-team-techsmith-fn
# #185472698 Bug fix data loading and state management using Hooks

## What does this PR do?

This PR fixes data fetching and state management across category and wishlist components, and improves route protection.

## Description of Task to be completed?

### Steps to reproduce
1. Fix the Google login functionality to match the functionalities of logging in using email and password.
2. Enhance the navigation behavior for logged-in users, redirecting them to the dashboard instead of the login page.
3. Fix the chat navigation bug.a
4. Fix the wishlist functionality.
5. Fix state management in categories component.

### Expected
All features should work as expected.

### Actual
+ Updates and fixes `/category` and `/wishlist` route.
+ All steps mentioned above works as expected except for Google Auth

## How should this be manually tested?

- Clone the repository using `git clone https://github.com/atlp-rwanda/e-comm-team-techsmith-fn`

- Checkout to the branch using `git checkout bg-navigations-185472698`

- Install node packages that the application depends on: `npm i`

- Create a `.env` file in the root directory and add all environment variables required to run the application. The required variables are listed in the `.env.example` file available in root directory of the application.

- Start the application development server: `npm start`

- Check out the components documentation: `npm run storybook`

## What are the relevant pivotal tracker/Trello stories?

+ [#185472698](https://www.pivotaltracker.com/story/show/185472698)

## Screenshots (if appropriate)

https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/037dbb64-fd1d-4d0f-92fe-19e6e5934db7
